src/Accelerators/NNPA/NNPAAccelerator.hpp

```diff
@@ -21,14 +21,22 @@ namespace accel {
 namespace nnpa {
 
 class NNPAAccelerator final : public Accelerator {
-private:
-  static bool initialized;
+  friend Accelerator;
 
 public:
-  NNPAAccelerator();
+  ~NNPAAccelerator() {}
+
+  /// Define classof to be able to use isa<>, cast<>, dyn_cast<>, etc.
+  static bool classof(const Accelerator *accel) {
+    return accel->getKind() == Accelerator::Kind::NNPA;
+  }
+  static bool classof(const NNPAAccelerator *) { return true; }
+
+private:
+  NNPAAccelerator() : Accelerator(Accelerator::Kind::NNPA) {}
```

Comment:
Constructor is private so that only `Accelerator::create` can construct its derived classes (I made that class a friend).